vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/MenuItemsArrayGenerator.java

```diff
@@ -62,11 +62,7 @@ public void generate() {
             getItems().forEach(this::resetContainers);
 
             int containerNodeId = createNewContainer(menu.getChildren());
-            String appId = ui.getInternals().getAppId();
-
-            ui.getPage().executeJavaScript(
-                    "window.Vaadin.Flow.contextMenuConnector.generateItems($0, $1, $2)",
-                    getElement(), appId, containerNodeId);
+            getElement().callJsFunction("$connector.generateItems", containerNodeId);
```

Comment:
**note:** Now that, the menu items generator invokes `$connector.generateItems` whose implementation slightly differs between the menu-bar and context-menu connectors.